cpp/src/icegriddb/IceGridDB.cpp

```diff
@@ -275,20 +275,17 @@ run(const Ice::StringSeq& args)
 
             if(!serverVersion.empty())
             {
-                auto factory = [serverVersion](const string& type) -> shared_ptr<Ice::Value>
-                {
-                    if(type == "::IceGrid::ServerDescriptor")
+                communicator->getValueFactoryManager()->add([serverVersion](const string& type)
                     {
+                        assert(type == ServerDescriptorI::ice_staticId());
```

Comment:
Shouldn't this be 
```
assert(type == ServerDescriptor::ice_staticId());
```
No I suffix in ServerDescriptor

Reply:
You need the `IceGrid::` namespace there

```
@@ -277,15 +277,15 @@ run(const Ice::StringSeq& args)
             {
                 communicator->getValueFactoryManager()->add([serverVersion](const string& type)
                     {
-                        assert(type == ServerDescriptor::ice_staticId());
+                        assert(type == IceGrid::ServerDescriptor::ice_staticId());
                         return make_shared<ServerDescriptorI>(serverVersion);
-                    }, ServerDescriptor::ice_staticId());
+                    }, IceGrid::ServerDescriptor::ice_staticId());

                 communicator->getValueFactoryManager()->add([serverVersion](const string& type)
                     {
-                        assert(type == IceBoxDescriptor::ice_staticId());
-                        return make_shared<IceBoxDescriptor>(serverVersion);
-                    }, IceBoxDescriptor::ice_staticId());
+                        assert(type == IceGrid::IceBoxDescriptor::ice_staticId());
+                        return make_shared<IceBoxDescriptorI>(serverVersion);
+                    }, IceGrid::IceBoxDescriptor::ice_staticId());
             }```

Reply:
Pushed a fix, thanks!